### PR TITLE
Set all reader mode content as text instead of HTML

### LINKED ISSUE #4772: [hackerone] reader mode flaw
https://hackerone.com/reports/1436142

credit: nishimunea

### Test Plan

* https://github.com/brave/internal/issues/843

### PR DESCRIPTION
**Pull request by diracdeltas**
and set nonce before replacing other text

## Summary of Changes

This pull request fixes #4772 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

STR in hackerone ticket

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Reader/Reader.html [hunk 1]
```diff
@@ -15,20 +15,14 @@
...
+    <div id="reader-credits" class="credits"></div>
     <script type="text/javascript" src="/reader-mode/javascript/Reader.js"></script>
-    <script nonce="%READER-TITLE-NONCE%">
-        document.getElementById("reader-page-title").textContent = "%READER-TITLE%";
-        document.getElementById("reader-title").textContent = "%READER-TITLE%";
-    </script>
   </div>
 
   <div id="reader-content" class="content">
-    %READER-CONTENT%
```
Comment (diracdeltas) [inline Client/Frontend/Reader/Reader.html:27]: note that this fix may break some reader mode pages if `READER-CONTENT` is actually supposed to contain HTML

File: Client/Frontend/Reader/Reader.html [hunk 2]
```diff
@@ -15,20 +15,14 @@
...
   </div>
 
   <div id="reader-message" class="message">
-    %READER-MESSAGE%
```
Comment (diracdeltas) [inline Client/Frontend/Reader/Reader.html:31]: same comment as https://github.com/brave/brave-ios/pull/4773/files#r775208107

**Review by Brandon-T — APPROVED** [review]

File: Client/Frontend/Reader/ReaderModeHandlers.swift [hunk 3]
```diff
@@ -44,14 +45,10 @@ struct ReaderModeHandlers {
...
-                        let setTitleNonce = UUID().uuidString.replacingOccurrences(of: "-", with: "")
-                        
-                        if let html = ReaderModeUtils.generateReaderContent(readabilityResult, initialStyle: readerModeStyle,
-                            titleNonce: setTitleNonce),
+                        if let html = ReaderModeUtils.generateReaderContent(readabilityResult, initialStyle: readerModeStyle),
                             let response = GCDWebServerDataResponse(html: html) {
                             // Apply a Content Security Policy that disallows everything except images from anywhere and fonts and css from our internal server
-                            response.setValue("default-src 'none'; img-src *; style-src http://localhost:* '\(readerModeStyleHash)'; font-src http://localhost:*; script-src 'nonce-\(setTitleNonce)'", forAdditionalHeader: "Content-Security-Policy")
+                            response.setValue("default-src 'none'; img-src *; style-src http://localhost:* '\(readerModeStyleHash)'; font-src http://localhost:*; script-src '\(readerModeScriptHash)'", forAdditionalHeader: "Content-Security-Policy")
```
Comment (Brandon-T) [inline Client/Frontend/Reader/ReaderModeHandlers.swift]: When ran live, the dynamic script won't run as the hash is static and the content of the script is replaced at runtime. For some reason in Chrome Desktop and Brave Desktop it works fine, but not on iPad and my 14.6 iPhone.

Looking into putting the JS in a separate file atm.

File: Client/Frontend/Reader/Reader.html [hunk 4]
```diff
@@ -46,6 +40,13 @@ <h1 id="reader-title"></h1>
     <li><a id="toggle-button" class="button toggle-button" href="#"></a></li>
   </ul>
 
+  <script>
+      document.getElementById("reader-page-title").textContent = "%READER-TITLE%";
+      ["title", "credits", "content", "message"].forEach((name) => {
+        document.getElementById(`reader-${name}`).textContent = `%READER-${name.toUpperCase()}`;
```
Comment (Brandon-T) [inline Client/Frontend/Reader/Reader.html:46]: Just realized some pages render as HTML. Everything else can be `textContent` except for `reader-content` which can render HTML via `innerHTML`.

Still reviewing to see if any other websites break it.

**Review by Brandon-T — CHANGES_REQUESTED** [review]

File: Client/Frontend/Reader/ReaderModeUtils.swift [hunk 5]
```diff
@@ -20,14 +20,29 @@ struct ReaderModeUtils {
...
-            .replacingOccurrences(of: "%READER-CREDITS%", with: readabilityResult.credits)
+            .replacingOccurrences(of: "%READER-TITLE%", with: readabilityResult.title.javaScriptEscapedString?.unquotedIfNecessary ?? readabilityResult.title.htmlEntityEncodedString)
+            .replacingOccurrences(of: "%READER-CREDITS%", with: readabilityResult.credits.javaScriptEscapedString?.unquotedIfNecessary ?? readabilityResult.credits.htmlEntityEncodedString)
             .replacingOccurrences(of: "%READER-CONTENT%", with: readabilityResult.content)
-            .replacingOccurrences(of: "%READER-TITLE-NONCE%", with: titleNonce)
+            .replacingOccurrences(of: "%READER-MESSAGE%", with: "")
+    }
+}
 
+private extension String {
```
Comment (diracdeltas) [inline Client/Frontend/Reader/ReaderModeUtils.swift:36]: possible to add some tests for this?

**Review by Brandon-T — APPROVED** [review]

File: Client/Frontend/Reader/ReaderModeUtils.swift (hunk 5 above)
Comment (iccub) [reply Client/Frontend/Reader/ReaderModeUtils.swift:36]: Left unit tests to handle later https://github.com/brave/brave-ios/issues/4794

**Pull request merged**